cmd/thanos/query_frontend.go

```diff
@@ -148,6 +149,8 @@ func registerQueryFrontend(app *extkingpin.App) {
 
 	cmd.Flag("query-frontend.log-failed-queries", "Log failed queries due to any reason").Default("true").BoolVar(&cfg.CortexHandlerConfig.LogFailedQueries)
 
+	cmd.Flag("failed-query-cache-capacity", "Capacity of cache for failed queries").IntVar(&cfg.CortexHandlerConfig.FailedQueryCacheCapacity)
```

Comment:
nit: give it a default value to 0 which means it is disabled by default.

cmd.Flag("failed-query-cache-capacity", "Capacity of cache for failed queries. 0 means this feature is disabled.").IntVar(&cfg.CortexHandlerConfig.FailedQueryCacheCapacity).Default(0)

Reply:
is preferred syntax for intvar default as you have specified here or as shown in the following flag

cmd.Flag("labels.max-query-parallelism", "Maximum number of labels requests will be scheduled in parallel by the Frontend.").Default("14").IntVar(&cfg.LabelsConfig.Limits.MaxQueryParallelism)